ext/mbstring/tests/gh10627.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+GH-10627 (mb_convert_encoding crashes PHP on Windows)
+--EXTENSIONS--
+mbstring
+--FILE--
+<?php
+
+$str = 'S�kinst�llningar';
+$data = [$str, 'abc'];
+var_dump(mb_convert_encoding($data, 'UTF-8', 'auto'));
+$data = [$str => 'abc', 'abc' => 'def'];
+var_dump(mb_convert_encoding($data, 'UTF-8', 'auto'));
+
```

Comment:
I reproduced  `--enable-debug` on WSL on Ubuntu and it is seems C NULL problem. But [3v4l](https://3v4l.org/gPqjF) looks works fine and result is wrong because maybe without `--enable-debug` option. 
@alexdowad what do you think?

Reply:
Hmm. That is interesting. I'm not sure of the reason right now.

This bug occurs if `'auto'` is passed as the desired encoding, and mbstring is not able to detect the encoding of the string.

I might need to check under which circumstances that can happen. In any case, this bug fix is definitely needed.